Feature/display all approved job listings on the homepage

ivailo-devrix opened the pull request.

ivailo-devrix committed:
styling/add-logo

add logo
change index.html to php

ivailo-devrix committed:
Update .gitignore

ivailo-devrix committed:
update .gitignore

ivailo-devrix committed:
architecture/new-file-structure

Move css to theme folder

ivailo-devrix committed:
architecture/renaming-a-directory

from themes to theme

ivailo-devrix committed:
architecture/rename-html-to-php-split-header-and-footer

ivailo-devrix committed:
architecture/delete-header.tpl

ivailo-devrix committed:
update folders structure

ivailo-devrix committed:
Update view-submission.php

ivailo-devrix committed:
Display all approved job listings on the homepage

Display all approved job listings on the homepage + adding sample images

Neychok commented on includes/db-connect.php:
```diff
@@ -0,0 +1,40 @@
+
+    $result = mysqli_query($conn, $sql);
+
+    close_db_conn($conn);
```
Hey, let's not close the connection on every SQL query.

ivailo-devrix replied:
I deleted the auto DB close from the function.

Neychok commented on includes/theme-compat/header.php:
```diff
@@ -0,0 +1,52 @@
+<?php
+//require_once (dirname(__FILE__).'/config/config.php');
+require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/config.php');
+require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/functions.php');
+require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/db-connect.php');
```
You actually might want to require the db-connect.php file before the functions.php file
```suggestion
require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/db-connect.php');
require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/functions.php');
```

ivailo-devrix replied:
I changed the calling position.

Neychok commented on includes/config.php:
```diff
@@ -0,0 +1,16 @@
+<?php
+const SSL_ENABLED = true;
+const BASE_URL = '/test/';
+
+const DB_NAME = 'joobs_v_3_0';
+const DB_USER = 'root';
+const DB_PASSWORD = '';
+const DB_HOST = 'localhost';
+
+const IMAGE_PATH = 'img/company-logo/';
+const DOC_UPLOAD_PATH = 'uploads/';
```
Could you please use `define()` instead of `const`

ivailo-devrix replied:
Fixed

Neychok commented on includes/urls.php:
```diff
@@ -0,0 +1,14 @@
+<?php
+
+// getting the real domain where the site is hosted
+$domain = $_SERVER['SERVER_NAME'];
+
+
+// getting HTTPS parameter if SSL is turned on
+if (SSL_ENABLED){
+    $httpParam = "https://";
+}else{
+    $httpParam = "http://";
+}
+
+$HTTP_DOMAIN = $httpParam.$domain ;
```
I am not liking this whole file. I can see that you already have a define for the base URL of the site, so why not use it :)

Neychok commented on includes/theme-compat/header.php:
```diff
@@ -0,0 +1,52 @@
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+
+
+    <?php echo "<link rel=\"stylesheet\" href=\"$HTTP_DOMAIN/assets/css/master.css\">"; ?>
```
I already mentioned it in another comment, but let's use the `BASE_URL` define instead of `$HTTP_DOMAIN`

Neychok commented on index.php:
```diff
@@ -0,0 +1,204 @@
+<?php
+require_once(dirname(__FILE__) . '/includes/theme-compat/header.php');
+require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/db-connect.php');
```
One time you use `dirname` then right below you use `$_SERVER['DOCUMENT_ROOT']`. Is there a reason for that?

ivailo-devrix replied:
Fixed

Neychok reviewed (changes requested):
Hey, the PR is about displaying the job listings on the homepage but instead I get 60 files unrelated to the PR's original purpose. I get why that might have happen but let's try not to do that in the future.

Also, could you please remove the images from the PR, as we don't want those inside master.

ivailo-devrix committed:
Update .gitignore

ivailo-devrix committed:
remove closing the connection on every SQL query

ivailo-devrix committed:
Update index.php

 require the db-connect.php file before the functions.php file
fix  replace $HTTP_DOMAIN whit  BASE_URL

ivailo-devrix committed:
use define() instead of const

ivailo-devrix committed:
rename url.php to functions.php

- rename url.php to functions.php
- remove unused functions
- add date_default_timezone_set

ivailo-devrix committed:
change dirname(__FILE__)   to  $_SERVER['DOCUMENT_ROOT']

ivailo-devrix committed:
Update .gitignore

add img/company/

ivailo-devrix committed:
delete company logos image

ivailo-devrix committed:
replace $HTTP_DOMAIN whit BASE_URL

Neychok commented on actions-job.php:
```diff
@@ -31,4 +31,6 @@
-<?php require_once (dirname(__FILE__) . '/includes/theme-compat/footer.php');  ?>
\ No newline at end of file
+<?php
+require_once($_SERVER['DOCUMENT_ROOT'] . '/includes/theme-compat/footer.php');
```
```suggestion
require_once(ABSPATH. 'includes/theme-compat/footer.php');
```

Neychok reviewed (changes requested):
One last thing and this is ready.

Neychok reviewed (approved):
Looks good.
